sdk-cpp/CMakeLists.txt

```diff
@@ -10,40 +8,126 @@ set(DO_PLATFORM_ID_MAC 2)
+    # For now, linux support of the SDK is limited to only the open-sourced agent.
+    set(DO_CLIENT_ID ${DO_CLIENT_ID_AGENT})
+elseif (CMAKE_SYSTEM_NAME MATCHES Darwin)
+    set(DO_CLIENT_ID ${DO_CLIENT_ID_DOSVC})
+elseif (CMAKE_SYSTEM_NAME MATCHES Windows)
+    set(DO_CLIENT_ID ${DO_CLIENT_ID_DOSVC})
+else()
+    message(WARNING "Unknown platform - client not supported on this platform")
+    set(DO_CLIENT_ID_UNKNOWN ${DO_PLATFORM_ID_UNKNOWN})
+endif()
+message(STATUS "DO_CLIENT_ID: ${DO_CLIENT_ID}")
+
+if (${DO_PLATFORM_ID} EQUAL ${DO_PLATFORM_ID_LINUX} OR ${DO_PLATFORM_ID} EQUAL ${DO_PLATFORM_ID_MAC}) 
+    message("DO configured to use shared Boost libraries")
+    add_definitions(-DBOOST_ALL_DYN_LINK=1)
+endif()
+
+if (${DO_PLATFORM_ID} EQUAL ${DO_PLATFORM_ID_LINUX})
+    fixup_compile_options_for_arm()
+endif()
 
 # Include external libraries here
 find_package(Boost COMPONENTS filesystem system REQUIRED)
+if(${DO_PLATFORM_ID} EQUAL ${DO_PLATFORM_ID_MAC})
+    find_package(Microsoft.GSL CONFIG REQUIRED)
+endif()
 
 if (DO_BUILD_TESTS)
     add_subdirectory(tests)
 endif()
 
-file(GLOB sdk_source
+set(DO_SDK_LIB_NAME "deliveryoptimization")
+add_do_version_lib("${DO_SDK_LIB_NAME}-lib" ${PROJECT_VERSION})
+
+### Region CMake Build ###
+
+set(sdk_source_common
     "src/*.cpp"
-    "src/internal/rest/*.cpp"
-    "src/internal/util/*.cpp"
+    "src/internal/*.cpp"
 )
 
-set(DO_SDK_LIB_NAME "deliveryoptimization")
+set(sdk_compile_definitions_common
+    DO_INTERFACE_ID=${DO_INTERFACE_ID}
+    DO_INTERFACE_ID_REST=${DO_INTERFACE_ID_REST}
+    DO_INTERFACE_ID_COM=${DO_INTERFACE_ID_COM}
+    DO_INTERFACE_ID_UNKNOWN=${DO_INTERFACE_ID_UNKNOWN}
+    DO_CLIENT_ID=${DO_CLIENT_ID}
+    DO_CLIENT_ID_DOSVC=${DO_CLIENT_ID_DOSVC}
+    DO_CLIENT_ID_AGENT=${DO_CLIENT_ID_AGENT}
+    DO_CLIENT_ID_UNKNOWN=${DO_CLIENT_ID_UNKNOWN}
+)
+
+set(sdk_public_includes
+    $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${DO_SDK_LIB_NAME}>
+)
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+set(sdk_private_includes_common
+    "src/internal"
+)
+
+if(${DO_PLATFORM_ID} EQUAL ${DO_PLATFORM_ID_LINUX})
     include(GNUInstallDirs)
 
-    add_do_version_lib("${DO_SDK_LIB_NAME}-lib" ${PROJECT_VERSION})
+    file(GLOB sdk_source
+        ${sdk_source_common}
+        "src/internal/rest/*.cpp"
+        "src/internal/rest/util/*.cpp"
+    )
 
     add_library(${DO_SDK_LIB_NAME} SHARED "${sdk_source}")
+    
+    set(sdk_compile_definitions
+        DOSVC_BIN_NAME="${DOSVC_BIN_NAME}"
+        DO_PLUGIN_APT_BIN_NAME="${DO_PLUGIN_APT_BIN_NAME}"
+    )
+
+    set(sdk_private_includes
+        "src/internal/rest"
+        "src/internal/rest/util"
+        ${include_directories_for_arm}
+    )
+
+    set(sdk_private_linked_libs
+        doversion
+        dohttp
+    )
+
+    set(sdk_public_linked_libs
+        ${Boost_LIBRARIES}
+    )
+
     add_library(Microsoft::${DO_SDK_LIB_NAME} ALIAS ${DO_SDK_LIB_NAME})
```

Comment:
>Microsoft:: [](http://example.com/codeflow?start=16&length=11)

Use the namespace on mac and windows as well?

Reply:
Nevermind. Please delete this line as well since it is a dupe of line 183.

Reply:
Reason you retained this line?

Reply:
meant to delete